### ISSUE TITLE
Doesn't support Feature Layers populated with Feature Collections instead of URL

### ISSUE THREAD

**Issue by kevinsigwart** [swingley/cereal#8]
If there is a Feature Layer added to the map that is a feature collection, the graphics do not get serialized.

For example, if a layer was was added...
 featureLayer = new FeatureLayer(featureCollection, {
                  id: 'exampleLayerID',
                  name: 'My Example Layer',
                  infoTemplate: popupTemplate
                });
map.addLayers([featureLayer])

Then when the serializing occurs the code does not account for FeatureLayer's that have graphics and are not referenced via URL.

**Comment by swingley** [comment]
Yes, that's mentioned in a comment:  https://github.com/swingley/cereal/blob/2bffbcf1e4d178e2c6bfef3af1e6c05c65d09501/extras/map-cereal.js#L28

I think what you're asking for can be added easily. I'll see what I can do to add it.

**Comment by kevinsigwart** [comment]
Hi Derek,

   I started to add it on my branch.  I can send you a pull request.  I
check to see if there is a URL, if not, I serialize the feature
collection.  I ran into some issues with the symbology though.  I just
didn't have the time to serialize all the symbology combinations aka Unique
Values, Simple Polygons, Simple Points...  I ended up just hard coding a
default symbology.

   I'll send you a pull request shortly.

Thanks for getting back to me.

Kevin

**Comment by swingley** [comment]
Thanks for the PR. I'd rather not have hard-coded symbology and support for a single geometry type in this. Instead I added support for feature layers from feature collections. My code is in the [feature-layer-from-feature-collection](https://github.com/swingley/cereal/tree/feature-layer-from-feature-collection) branch. Do you have time to test with that branch and see if it works for your use case?

**Comment by kevinsigwart** [comment]
Yeah, i sort of figured that.  I just wrote it in a way that would suit my
needs for my application and just didn't have time to write a more flexible
implementation.  I'll test it early next week, when I'm back in the
office.

Thanks!
